### PR TITLE
docs: add components overview page

### PR DESCRIPTION
**Pull request by bbogdanov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by hippee-lee — CHANGES_REQUESTED** [review]
Looking good and on track with what we have been discussing. 

I'm blocking for two reasons:

1. There is still a problem we need to solve for when navigationg to the component overview page - iir this might be handled in the Sidebar.vue component.
2. We should also include the svgs for each component from the Figma file with this PR. I think they are available for export from the figma file.

**Review by hippee-lee — APPROVED** [review]

File: apps/website/.vuepress/theme/global-components/ItemOverview.vue [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  </section>
+</template>
+
+<script>
+export default {
+  name: 'ItemOverview',
+  props: ['title', 'path', 'target', 'isNew'],
+  methods: {
+    navigate: function () {
+      window.location = this.path;
```
Comment (Shijir) [inline apps/website/.vuepress/theme/global-components/ItemOverview.vue:22]: Can we do it with Vue router here?

Comment (hippee-lee) [reply apps/website/.vuepress/theme/global-components/ItemOverview.vue:22]: The Vue router is available in the theme components. I think it is provided and accessed via `this.$route`.
@shijir didn't block so I'm assuming he considers it a nice to have and not a blocking issue to these changes.

**Comment by gnomeontherun** [comment]
Can we please add this to the navigation?

**Comment by gnomeontherun** [comment]
Did we check if design is happy? I see the height of some cards are different, would like consistency perhaps.

**Comment by bbogdanov** [comment]
> Did we check if design is happy? I see the height of some cards are different, would like consistency perhaps.

Here is an example of what happens when we make the image size consistent with CSS.

image

**Review by Shijir — APPROVED** [review]

File: apps/website/.vuepress/theme/global-components/ItemOverview.vue [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  </section>
+</template>
+
+<script>
+export default {
+  name: 'ItemOverview',
+  props: ['title', 'path', 'target', 'isNew'],
+  methods: {
+    navigate: function () {
+      this.$router.go(this.path);
```
[the remark this answers is not in the document]
Comment (bbogdanov) [reply apps/website/.vuepress/theme/global-components/ItemOverview.vue:22]: I didn't know that Vuepress used it and provides it in the root. I went through the entire documentation and there was no sign about the framework providing the router and also how to use it. God bless `console.log`

File: apps/website/.vuepress/theme/global-components/ComponentsOverview.vue [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      if (this.$site && this.$site.themeConfig && this.$site.themeConfig.sidebar) {
+        const componentsSection = this.$site.themeConfig.sidebar.find(x => x.title === 'Components');
+
+        if (componentsSection) {
+          const compSectionCopy = componentsSection.children.slice(); // create a copy of the array
+
+          compSectionCopy.splice(0, 1); // remove the Overview link
+
+          return compSectionCopy;
+        }
```
Comment (Shijir) [inline apps/website/.vuepress/theme/global-components/ComponentsOverview.vue:36]: Why not directly return `componentsSection.children.slice(1)`?

**Review by Shijir — DISMISSED** [review]

**Review by hippee-lee — APPROVED** [review]

**Review by hippee-lee — APPROVED** [review]

**Review by Shijir — APPROVED** [review]

**Pull request merged**